libcxx/include/__ranges/to.h

```diff
@@ -109,8 +109,11 @@ template <class _Container, input_range _Range, class... _Args>
         __result.reserve(static_cast<range_size_t<_Container>>(ranges::size(__range)));
       }
 
-      for (auto&& __ref : __range) {
-        using _Ref = decltype(__ref);
+      auto __iter = ranges::begin(__range);
```

Comment:
I would suggest we use `ranges::for_each` instead, that seems cleaner and I don't think saving an include dependency is worth the added code complexity.

Reply:
I also wonder whether this would improve interactions with `deque` iterators.

Reply:
Switched to use `ranges::for_each`. I wonder whether range-for on `ref_view{__range}` suffices.

Reply:
This is somehow bad, as encountered in https://github.com/llvm/llvm-project/pull/113103#discussion_r1808014474.